Scripts/Runtime/Core/Util/MathExtension.cs

```diff
@@ -0,0 +1,406 @@
+        /// True (component-wise) if the delta between <see cref="a"/> and <see cref="b"/> is less than
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </returns>
+        /// <remarks>
+        /// Resolves infinite and NaN comparisons as false with improved performance vs <see cref="IsApproximately"/>.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool3x3 IsApproximately(this float3x3 a, float3x3 b)
+        {
+            return new bool3x3(
+                a.c0.IsApproximately(b.c0),
+                a.c1.IsApproximately(b.c1),
+                a.c2.IsApproximately(b.c2)
+            );
+        }
+
+        /// <summary>
+        /// Checks if two <see cref="float4x4"/> values are approximately equal according to the
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </summary>
+        /// <param name="a">The <see cref="float4x4"/> to compare.</param>
+        /// <param name="b">The <see cref="float4x4"/> to compare.</param>
+        /// <returns>
+        /// True (component-wise) if the delta between <see cref="a"/> and <see cref="b"/> is less than
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </returns>
+        /// <remarks>
+        /// Resolves infinite and NaN comparisons as false with improved performance vs <see cref="IsApproximately"/>.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool4x4 IsApproximately(this float4x4 a, float4x4 b)
+        {
+            return new bool4x4(
+                a.c0.IsApproximately(b.c0),
+                a.c1.IsApproximately(b.c1),
+                a.c2.IsApproximately(b.c2),
+                a.c3.IsApproximately(b.c3)
+            );
+        }
+
+        /// <summary>
+        /// Checks if two <see cref="float"/> values are approximately equal according to the
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </summary>
+        /// <param name="a">The <see cref="float"/> to compare.</param>
+        /// <param name="b">The <see cref="float"/> to compare.</param>
+        /// <returns>
+        /// True if the delta between <see cref="a"/> and <see cref="b"/> is less than
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </returns>
+        /// <remarks>
+        /// Resolves infinite and NaN comparisons as true with the same logic as <see cref="IsEqualOrNaN"/> at
+        /// additional cost vs <see cref="IsApproximately"/>.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsApproximatelySafe(this float a, float b)
+        {
+            return (math.abs(a - b) < MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE)
+                   | (math.isnan(a) & math.isnan(b))
+                   | (math.isinf(a) & math.isinf(b) & a == b);
```

Comment:
Is the a == b check because `isinf` will return positive or negative infinity?

Reply:
That's right